Sources/StreamChatUI/Utils/Reusable+Extensions.swift

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+protocol Reusable {
+    static var reuseIdentifier: String { get }
+}
+
+extension Reusable {
+    static var reuseIdentifier: String {
+        String(describing: self)
+    }
+}
+
+// swiftlint:disable force_cast
+extension UITableViewCell: Reusable {}
+
+extension UITableView {
+    /// Registers a `UITableViewCell` from a supplied type
+    /// The identifier is used from the reuseIdentifier parameter
+    /// - Parameter type: A generic cell type
+    func register<Cell: UITableViewCell>(_ type: Cell.Type) {
+        register(type, forCellReuseIdentifier: type.reuseIdentifier)
+    }
+    
+    /// Dequeues a `UITableView` cell with a generic type and indexPath
+    /// - Parameters:
+    ///   - type: A generic cell type
+    ///   - indexPath: The indexPath of the row in the UITableView
+    /// - Returns: A cell from the type passed through
+    public func dequeueReusableCell<Cell: UITableViewCell>(with type: Cell.Type, for indexPath: IndexPath) -> Cell {
+        dequeueReusableCell(withIdentifier: type.reuseIdentifier, for: indexPath) as! Cell
+    }
```

Comment:
I'm not sure having reusable helpers in SDK is a good idea as it can lead to collision with other 3rd party/customer's implementation. I was thinking of adding this just to the `DemoApp` (the same for storyboard helpers)